src/app/features/data-import/data-import.service.spec.ts

```diff
@@ -152,4 +153,48 @@ describe("DataImportService", () => {
   it("should put csv into db", async () => {
     // Todo, missing importCsv Function
   });
+
+  it("should validate csv with matching _id and content", async () => {
+    const mockFileReader = createFileReaderMock("_id,value\nChild:123,123");
+    const valResult = await service.validateCsvFile(null, "Child");
+
+    expect(mockFileReader.readAsText).toHaveBeenCalled();
+    expect(valResult.status).toEqual(CsvValidationStatus.Valid);
+  });
+
+  it("should validate csv without _id", async () => {
+    const mockFileReader = createFileReaderMock("value\n123");
+    const valResult = await service.validateCsvFile(null, "Child");
+
+    expect(mockFileReader.readAsText).toHaveBeenCalled();
+    expect(valResult.status).toEqual(CsvValidationStatus.Valid);
+  });
+
+  it("should reject csv when _id and content don't match", async () => {
+    const mockFileReader = createFileReaderMock("_id,value\nNotAChild:123,123");
+    const valResult = await service.validateCsvFile(null, "Child");
+
+    expect(mockFileReader.readAsText).toHaveBeenCalled();
+    expect(valResult.status).toEqual(CsvValidationStatus.ErrorWrongType);
+  });
+
+  it("should reject empty csv file", async () => {
+    const mockFileReader = createFileReaderMock("_id,value");
+    const valResult = await service.validateCsvFile(null, "Child");
+
+    expect(mockFileReader.readAsText).toHaveBeenCalled();
+    expect(valResult.status).toEqual(CsvValidationStatus.ErrorEmpty);
+  });
+
+  it("should reject empty csv without _id", async () => {
+    const mockFileReader = createFileReaderMock("value");
+    const valResult = await service.validateCsvFile(null, "Child");
+
+    expect(mockFileReader.readAsText).toHaveBeenCalled();
+    expect(valResult.status).toEqual(CsvValidationStatus.ErrorEmpty);
+  });
+
+  it("should reject invalid csv", async () => {
+    // I'm unable to break the csv import
```

Comment:
```suggestion
   spyOn(TestBed.inject(Papa), "parse").and.throwError(new Error());
```

Reply:
That would fail the `this.papa.parse(...)` call so you can work with the error case.

Reply:
Thanks for the hint, gotta try that out :)